deephaven/web-client-ui
# feat: JS API reconnect

- Listen for `EVENT_DISCONNECT` and `EVENT_RECONNECT` on the connection, and display a "Reconnecting..." message in the console.
- No longer listen to the `HACK_CONNECTION_FAILURE` (it's been deprecated).
- Listen for the `SHUTDOWN` event and display a message after shutdown.
- Fixes #1140 
Testing steps:
1. Start up server with deephaven-core JS API reconnect changes: https://github.com/deephaven/deephaven-core/pull/3502
2. Use ngrok to start a tunnel to that port, e.g.: `ngrok http 10000`
3. Start up Web UI connecting to that tunnel, e.g.: `VITE_CORE_API_URL=http://acfc-23-233-0-34.ngrok.io/jsapi npm start`
4. Open the Web UI in Firefox, run some commands to make sure initial connection is fine.
5. Press Alt and from the File menu, select "Work Offline"
6. See it transition to a disconnected state. Disconnected message should appear and should not be able to enter new commands
7. Reconnect by deselecting "Work Offline" option from Step 5
8. See it transition to connected state. Should be able to run commands and have the results appear.
9. Kill the server (Ctrl+C). See it transition to a Shutdown state, app unloaded.
